src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,211 @@
+#:include "common.fypp"
+module stdlib_stats_distribution_PRNG
+    use stdlib_kinds
+    implicit none
+    private
+    integer, parameter :: MAX_INT_BIT_SIZE = 64
+    integer(int64), save :: st(4), si = 614872703977525537_int64
+    logical, save :: seed_initialized = .false.
+
+    public :: random_seed
+    public :: dist_rand
+    public :: jump
+    public :: long_jump
+
+
+    interface dist_rand
+    !! Version experimental
+    !!
+    !! Generation of random integers with different kinds
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure dist_rand_${t1[0]}$${k1}$
+        #:endfor
+    end interface dist_rand
+
+    interface random_seed
+    !! Version experimental
+    !!
+    !! Set seed value for random number generator
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+    !!
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure random_distribution_seed_${t1[0]}$${k1}$
+        #:endfor
+    end interface random_seed
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    function dist_rand_${t1[0]}$${k1}$(n) result(res)
+    !! Random integer generation for various kinds
+    !! result = [-2^k, 2^k - 1], k = 7, 15, 31, 63, depending on input kind
+    !! Result is used as bit model number instead of regular arithmetic number
```

Comment:
This sentence isn't clear to me.

Reply:
Originally, the function dist_rand is aimed for unsigned integer algorithm. For 64bit, you have random integer in range[0, 2^64-1], for 32bit, you have random integer in range[0, 2^32 -1], etc. However, Fortran does not have unsigned integer. The same algorithm results in signed integer in [-2^63, 2^63-1] for 64bit, etc. Fortunately, the procedures of producing real random numbers later on will have bitwise operators instead of arithmetic operators. Therefore, algorithm is still valid, but the description is kind of awkward.